custom_components/tuya_local/devices/alecoair_d12_home_dehumidifier.yaml

```diff
@@ -0,0 +1,235 @@
+name: AlecoAir D12 Home dehumidifier
+products:
+  - id: hp1sjssyokokbrsg
+#    name: AlecoAir D12 Home
+primary_entity:
+  entity: humidifier
+  class: dehumidifier
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+      mapping:
+        - dps_val: false
+          icon: "mdi:power"
+          icon_priority: 1
+        - dps_val: true
+          icon: "mdi:power"
+          icon_priority: 4
+    - id: 2
+      name: mode
+      type: integer
+      mapping:
+        - dps_val: Continuous
+          value: 1
+          icon: mdi:fan
+          icon_priority: 3
+        - dps_val: Auto
+          value: 0
+          icon: "mdi:fan-auto"
+          icon_priority: 3
+    - id: 4
+      name: humidity
+      type: integer
+      range:
+        min: 30
+        max: 80
+      mapping:
+        - step: 5
+    - id: 11
+      name: error
+      type: bitfield
+      # Docs mention E1 and E2, it isn't clear which is tank full, so for now
```

Comment:
It should be fairly easy to find which error code is related to the tank.  Normally the same error will trigger when you remove the tank.  To check it is the same error code, you can fill the tank up before you put it back in (or see what error appears in the app).